src/charm.py

```diff
@@ -1507,6 +1660,7 @@ def _generate_relation_departed_key(rel_id: int) -> str:
     @property
     def _is_removing_last_replica(self) -> bool:
         """Returns True if the last replica (juju unit) is getting removed."""
+        logger.error(f"{self.app.planned_units() = } | {len(self.peers_units) = }")
```

Comment:
I think this can be removed. If you disagree can you add some more context to it and make it a logger.info?

Reply:
Same it's a miss from my side :)